### ISSUE TITLE
Support openapi specification for Tuple fields

### ISSUE THREAD

**Issue by victorbenichoux** [pydantic/pydantic#3210]
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
pydantic version: 1.8.2
pydantic compiled: True
python version: 3.7.11 (default, Jul 13 2021, 09:42:02)  [Clang 12.0.0 (clang-1200.0.32.29)]
platform: Darwin-19.6.0-x86_64-i386-64bit
optional deps. installed: ['typing-extensions']
```

When creating openapi schemas, `BaseModel` fields that are `Tuple` are indistinguishable from `List` instances. This is despite the fact that the openapi specification [now supports fixed-length lists (i.e. tuple-like)](https://json-schema.org/understanding-json-schema/reference/array.html#tuple-validation).

Specifically, the following here is true:
```py
import pydantic
from typing import List, Tuple

class ListModel(pydantic.BaseModel):
    items: List[str]

class TupleModel(pydantic.BaseModel):
    items: Tuple[str]
    
list_model_schema = ListModel.schema()
tuple_model_schema = TupleModel.schema()

assert list_model_schema["properties"] == tuple_model_schema["properties"]
```

This leads to an issue with the openapi specification generated by fastapi, in which the swagger can not correctly render the content of `Tuple` members, as referenced in these issues https://github.com/tiangolo/fastapi/issues/3665 and https://github.com/tiangolo/fastapi/issues/3782, and this pull request https://github.com/tiangolo/fastapi/pull/3874

# Suggestion

I think that it would make sense to support the openAPI specification for `Tuple` types, as a result, the specification of the
 `TupleModel` would change from
```json
{
  "title": "TupleModel",
  "type": "object",
  "properties": {
    "items": {
      "title": "Items",
      "type": "array",
      "items": { "type": "string" }
    }
  },
  "required": ["items"]
}
```
to
```json
{
  "title": "TupleModel",
  "type": "object",
  "properties": {
    "items": {
      "title": "Items",
      "type": "array",
      "prefixItems": [{ "type": "string" }],
      "items": false
    }
  },
  "required": ["items"]
}
```

Where, each field of the `Tuple` is specified by an entry in `prefixItems` (see the [docs](https://json-schema.org/understanding-json-schema/reference/array.html#tuple-validation)). In itself, this would allow extra arguments to the list, thus the fixed length nature of the `Tuple` is reflected by `items` being set to `false` (see [here](https://json-schema.org/understanding-json-schema/reference/array.html#additional-items)).

I have created a branch (https://github.com/victorbenichoux/pydantic/tree/fix-tuple) to implement this specification.

**Comment by victorbenichoux** [comment]
Something worth noting is that the `prefixItems` syntax for tuples is not available in the openapi spec `<3.1.0`, as a result this will break dependencies that only support `<=3.0`. This is the case of swagger for example (used, e.g. in fastapi). We may want to refrain from doing this until swagger supports it.

**Comment by Kludex** [comment]
This should be solved in V2.
